backend/src/stwo/prover.rs

```diff
@@ -430,12 +508,15 @@ where
         Ok(bincode::serialize(&proof).unwrap())
     }
 
-    pub fn verify(&self, proof: &[u8], _instances: &[M31]) -> Result<(), String> {
-        assert!(
-            _instances.is_empty(),
-            "Expected _instances slice to be empty, but it has {} elements.",
-            _instances.len()
-        );
+    pub fn verify(&self, proof: &[u8], instances: &[M31]) -> Result<(), String> {
+        // get public values
+        let public_values: BTreeMap<String, M31> = self
+            .analyzed
+            .get_publics()
+            .iter()
+            .zip_eq(instances.iter())
```

Comment:
This panics if the passed instances have the wrong size, instead we should error out cleanly before this iteration.